### PR TITLE
VACMS-12286 Add content_model_documentation module and enable.

### LINKED ISSUE #12286: Add informative field list
https://www.drupal.org/project/content_model_documentation/issues/3334953

The Drupal default field list (/admin/reports/fields) is not informative enough and could benefit from several improvements that would allow content model architects to see more information.

## ACs
- [ ] Modify or reproduce /admin/report/fields (preferably as a View)
Pulled from existing data
   - [x] Machine name (this will be the grouping)
   - [x] Field Type (with module that provides it)
   - [x] Entity Type (blocks, nodes, paragraphs,
      - [ ] ~Used in (link to manage fields)~ This was replaced by the combination of entity type and bundle. To account for cases where both an entity of block might be called the same thing as a content bundle.  Yes it is bad architecture, but they exist in the wild of other people's sites.  ;)
      - [x] Entity type
      - [x] bundle
      - [x] Field name
      - [x] Description
      - [x] target(s) (for entity references)
       - [x] cardinality
       - [x] required
      - [ ] ~link to display page for the field~ - updated The display page does not exist, so can not be linked at this time
- [ ] code attributed to [this issue](https://www.drupal.org/project/content_model_documentation/issues/3334953)

### PR DESCRIPTION
**Pull request by swirtSJW**
## Description

Closes #12286

## Testing done /QA steps

What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

As an admin  (this report only shows to admins)
1.  Login  https://pr12332-djcvxbj4y6ist5avej2dtuz3xylrdpu4.ci.cms.va.gov/ and pull down the reports menu, 
   - [ ] Validate that the "Content Model" appears in the menu under "Field list"  
   
image

 - [ ] Click the Content Model menu link, validate that it takes you to /admin/reports/fields/content-model
 - [ ] Validate that the View contains a "Displaying ..." counter above the filters and that it is showing ~800 items.
 - [ ] Validate "Content Model" is in the tabs for navigating to other Field related reports
 
image

  - [ ] Validate Field name  (Machine name)  is in results and filterable
   - [ ] Validate Field Type is in results and filterable
      - [ ] Validate entity reference targets are present in the Field Type column on entity reference-like fields
   - [ ] Entity Type is in results and filterable
   - [ ] Bundle is in results and filterable
   - [ ] ~Used in (link to manage fields)~ This was replaced by the combination of entity type and bundle.
   - [ ] Label (Field Label) is in results and filterable
   - [ ] Description is in results
   - [ ] cardinality is in results
   - [ ] required is in results
   - Views Style guide  This View is provided via a contrib module we control and as such is only loosely bound to our style guide, however I have tried to maintain as much it that makes sense.
      - [ ] All columns have headings (508)
      - [ ] Has table caption "List of fields used across the content model." (508)
      - [ ] Filter order matches column order
      - [ ] Buttons says "filter"
      - [ ] has a "reset" button
      - [ ] columns that can be sortable, are sortable
   
As content admin
- Login as Randi.Hecht@va.gov
- go to https://pr12332-djcvxbj4y6ist5avej2dtuz3xylrdpu4.ci.cms.va.gov/admin/reports/fields/content-model
- [ ] validate access denied  (we may open this up to other roles later, but  for now it is only admins.)

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] ~Tests have been added if necessary.~  This is largely contrib, we do not usually test contrib.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS Team`
- [ ] `Sitewide program`
- [x] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [x] No announcement is needed for this code change.
  - [x] Merge & carry on unburdened by announcements

### CODE REVIEW

File: config/sync/core.extension.yml [hunk 1]
```diff
@@ -30,10 +30,12 @@ module:
   config_ignore: 0
   config_override_warn: 0
   config_split: 0
+  config_views: 0
```
Comment (swirtSJW) [inline config/sync/core.extension.yml:33]: This module installs a bunch of Views  with config/install that override some of the core lists (that are not Views).  However they get removed on the next deploy step when config import runs the contents of config/sync.   This is intentional. I do not want them installing at this time as their scope and impact has not been determined.

**Comment by swirtSJW** [comment]
This is classic... I have to add a contrib View to the spec tool so that our spec tool tests pass.. for the very module that is meant to kill the spec tool.

**Comment by laflannery** [comment]
@swirtSJW Going through the QA steps I'm unclear what this one means "Used in (link to manage fields)".  Otherwise, everything else was clear and easy to follow/validate

**Comment by timcosgrove** [comment]
As an admin  (this report only shows to admins)
1.  Login  https://pr12332-djcvxbj4y6ist5avej2dtuz3xylrdpu4.ci.cms.va.gov/ and pull down the reports menu, 
   - [x] Validate that the "Content Model" appears in the menu under "Field list"  
   
image

 - [x] Click the Content Model menu link, validate that it takes you to /admin/reports/fields/content-model
 - [x] Validate that the View contains a "Displaying ..." counter above the filters and that it is showing ~800 items.
 - [ ] Validate "Content Model" is in the tabs for navigating to other Field related reports
 
image

  - [x] Validate Field name  (Machine name)  is in results and filterable
   - [x] Validate Field Type is in results and filterable
      - [ ] Validate entity reference targets are present in the Field Type column on entity reference-like fields
   - [ ] Entity Type is in results and filterable
   - [ ] Bundle is in results and filterable
   - [ ] Used in (link to manage fields)
   - [ ] Label (Field Label) is in results and filterable
   - [ ] Description is in results
   - [ ] cardinality is in results
   - [ ] required is in results
   - Views Style guide  This View is provided via a contrib module we control and as such is only loosely bound to our style guide, however I have tried to maintain as much it that makes sense.
      - [ ] All columns have headings (508)
      - [ ] Has table caption "List of fields used across the content model." (508)
      - [ ] Filter order matches column order
      - [ ] Buttons says "filter"
      - [ ] has a "reset" button
      - [ ] columns that can be sortable, are sortable
   
As content admin
- Login as Randi.Hecht@va.gov
- go to https://pr12332-djcvxbj4y6ist5avej2dtuz3xylrdpu4.ci.cms.va.gov/admin/reports/fields/content-model
- [ ] validate access denied  (we may open this up to other roles later, but  for now it is only admins.)

**Review by timcosgrove — DISMISSED** [review]
I did not dig through the Views Yaml, but this otherwise looks reasonable and everything that was asked for test-wise worked as expected.

**Comment by swirtSJW** [comment]
> @swirtSJW Going through the QA steps I'm unclear what this one means "Used in (link to manage fields)". Otherwise, everything else was clear and easy to follow/validate

Sorry my fault for not fixing the AC  "used in" is actually the reference to the entity type and the bundle.  I was originally mimic this from the field list
image

But then I found it not explicit enough to cover cases where we had a block named alert and a content type named alert or some other name collisions that might happen in other sites and not our own.

**Comment by swirtSJW** [comment]
image


I'll fix the conflict

**Review by laflannery — APPROVED** [review]

**Pull request merged**